### PR TITLE
Update openssl and perl

### PR DESCRIPTION
**Pull request by Lex-2008**

### CODE REVIEW

File: deps-packaging/openssl/solaris/build [hunk 1]
```diff
@@ -37,7 +43,7 @@ else
 fi
 
 $CONFIG shared  no-idea no-rc5 no-ssl2 no-ssl3 no-dtls no-psk no-srp no-asm \
-    --prefix=$PREFIX
+    $LXNET --prefix=$PREFIX
```
Comment (jimis) [inline deps-packaging/openssl/solaris/build]: Can you try adding it to the very end?

File: build-scripts/install-dependencies [hunk 2]
```diff
@@ -4,6 +4,25 @@
...
+echo '~~~ ensure this file is there'
+echo $LD_LIBRARY_PATH
+echo '~~~'
+find / -name libgcc_s.so.1 2>/dev/null || true
+echo '~~~ thanks'
+
+# test "x$label" == "xPACKAGES_sparc64_solaris_11" && export LD_LIBRARY_PATH=/usr/local/lib/
+export LD_LIBRARY_PATH=/usr/local/lib/
+# TODO: find a better place for this
+wget http://www.cpan.org/src/5.0/perl-5.24.1.tar.gz
```
Comment (jimis) [inline build-scripts/install-dependencies]: Why 5.24.1? Why not 5.24.3? Or even 5.26.1?

File: deps-packaging/README.md [hunk 3]
```diff
@@ -43,7 +43,7 @@ Hub specific dependencies:
 * [Apache](http://httpd.apache.org/) 2.4.27
 * [PostgreSQL](http://www.postgresql.org/) for the hub 9.6.3
 * [Redis](http://redis.io/) 3.2.9
-* [PHP](http://php.net/) 5.6.31
+* [PHP](http://php.net/) 7.2.0
 * [libmcrypt](https://sourceforge.net/projects/mcrypt/files/Libmcrypt/) 2.5.8
```
Comment (craigcomstock) [inline deps-packaging/README.md]: Ah! Searching for mcrypt in the PR found me this... probably remove as well right? nit.

File: build-scripts/install-dependencies [hunk 4]
```diff
@@ -25,6 +24,67 @@ install_pkg_files()
...
+#
+# By doing this, we rebuild on each and every platform. TODO create a proper
+# package for perl.
+#
+# NOTE: openssl also needs Module::Load::Conditional, which we install in the
+# cloud-init script (package libmodule-load-conditional-perl for debians and
+# perl-Module-Load-Conditional for RHELs).
+check_and_install_perl()
+{
+    rm -rf "$HOME/perl-my"                                 # TODO remove later
```
Comment (Lex-2008) [inline build-scripts/install-dependencies]: Time to remove, maybe?

File: build-scripts/install-dependencies [hunk 5]
```diff
@@ -25,6 +24,67 @@ install_pkg_files()
...
+
+    [ x"$PERL" = x ]  &&  [ -x "$HOME/perl-my/bin/perl" ]  \
+        && PERL="$HOME/perl-my/bin/perl"  \
```
Comment (Lex-2008) [inline build-scripts/install-dependencies]: $HOME/perl-my should not exist here anyway

File: deps-packaging/openssl/cfbuild-openssl.spec [hunk 6]
```diff
@@ -30,34 +30,49 @@ SYS=`uname -s`
...
-    ./config shared  no-idea no-rc5 no-ssl2 no-ssl3 no-dtls no-psk no-srp \
-        $DEBUG_CONFIG_FLAGS \
-        --prefix=%{prefix} \
-        $DEBUG_CFLAGS
+./config shared  no-idea no-rc5 no-ssl3 no-dtls no-psk no-srp no-engine \
+         $DEBUG_CONFIG_FLAGS \
+         --prefix=%{prefix} \
+         $HACK_FLAGS   \
+         $DEBUG_CFLAGS \
+	 $LDFLAGS
```
Comment (Lex-2008) [inline deps-packaging/openssl/cfbuild-openssl.spec]: TAB spotted

File: deps-packaging/openssl/mingw/debian/rules [hunk 7]
```diff
@@ -29,13 +36,23 @@ install: build
 	dh_clean -k
 	dh_installdirs
 
+	$(MAKE) install_sw DESTDIR=$(CURDIR)/debian/tmp
+	$(MAKE) install_ssldirs DESTDIR=$(CURDIR)/debian/tmp
+
+	find $(CURDIR)/debian/tmp
```
Comment (Lex-2008) [inline deps-packaging/openssl/mingw/debian/rules]: DEBUG TODO remove

File: deps-packaging/openssl/mingw/debian/rules [hunk 8]
```diff
@@ -29,13 +36,23 @@ install: build
...
+
 	mkdir -p $(CURDIR)/debian/tmp$(PREFIX)/bin
 	mkdir -p $(CURDIR)/debian/tmp$(PREFIX)/lib
 	mkdir -p $(CURDIR)/debian/tmp$(PREFIX)/include
 
-	cp libeay32.dll $(CURDIR)/debian/tmp$(PREFIX)/bin
+	pwd
+	ls -lahp
```
Comment (Lex-2008) [inline deps-packaging/openssl/mingw/debian/rules]: DEBUG TODO remove

File: deps-packaging/pkg-cache [hunk 9]
```diff
@@ -202,6 +202,8 @@ keep_newest)
     done
     cd -  >/dev/null
 
+    set -x
```
Comment (Lex-2008) [inline deps-packaging/pkg-cache]: @jimis : safe to remove now?

**Pull request merged**